### PR TITLE
Support parameter out in dpnp.add()

### PR DESCRIPTION
**Pull request by antonwolfy**
The PR implements support of `out` in-place parameter for `dpnp.add` function.
Additionally `__iadd__` function of `dpnp_array` class is added.

- [x] Have you provided a meaningful PR description?
- [x] Have you added a test, reproducer or referred to issue with a reproducer?
- [ ] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: tests/test_mathematical.py [hunk 1]
```diff
@@ -634,25 +635,75 @@ def test_invalid_shape(self, shape):
...
+            dpnp.add(dp_array1, dp_array2, out=dp_out)
+
+    @pytest.mark.parametrize("out",
+                             [4, (), [], (3, 7), [2, 4]],
+                             ids=['4', '()', '[]', '(3, 7)', '[2, 4]'])
+    def test_invalid_out(self, out):
+        a = dpnp.arange(10)
+
+        assert_raises(TypeError, dpnp.add, a, 2,  out)
+        assert_raises(TypeError, numpy.add, a.asnumpy(), 2,  out)
```
Comment (vlad-perevezentsev) [inline tests/test_mathematical.py:685]:
```suggestion
        assert_raises(TypeError, dpnp.add, a, 2, out)
        assert_raises(TypeError, numpy.add, a.asnumpy(), 2, out)
```

File: tests/test_mathematical.py [hunk 2]
```diff
@@ -634,25 +635,75 @@ def test_invalid_shape(self, shape):
             dpnp.trunc(dp_array, out=dp_out)
 
 
-class TestPower:
+class TestAdd:
+    @pytest.mark.parametrize("dtype", get_all_dtypes(no_none=True))
+    def test_power(self, dtype):
```
Comment (npolina4) [inline tests/test_mathematical.py:640]:
```suggestion
    def test_add(self, dtype):
```
Can you add tests for `__iadd__` function?

Comment (antonwolfy) [reply tests/test_mathematical.py:640]: > Can you add tests for `__iadd__` function?

It's already covered in `tests/test_usm_type.py`

File: tests/test_mathematical.py [hunk 3]
```diff
@@ -634,25 +635,75 @@ def test_invalid_shape(self, shape):
...
+        result = dpnp.add(dp_array1, dp_array2, out=dp_out)
 
-    def test_power(self):
+        # original
+        np_array1 = numpy.array(array1_data, dtype=dtype)
+        np_array2 = numpy.array(array2_data, dtype=dtype)
+        expected = numpy.add(np_array1, np_array2, out=out)
+
+        assert_allclose(expected, result)
+        assert_allclose(out, dp_out)
```
Comment (antonwolfy) [inline tests/test_mathematical.py:657]:
```suggestion
        assert_allclose(out, dp_out)

```

**Review by npolina4 — APPROVED** [review]

**Pull request merged**